### PR TITLE
[SQL-DS-CACHE-57][POAE7-969]replace raw array with std::vector

### PR DESCRIPTION
**Pull request by jikunshang**
## What changes were proposed in this pull request?

replace raw array with std::vector, rename some files to keep code style clean.

## How was this patch tested?
manual tests

### CODE REVIEW

**Comment by jikunshang** [comment]
@winningsix @xieqi @iyupeng Please take a review, thanks.

**Comment by iyupeng** [comment]
LGTM. 
A small question, could you tell us why replacing the raw arrays?

**Review by iyupeng — APPROVED** [review]

**Comment by jikunshang** [comment]
> LGTM.
> A small question, could you tell us why replacing the raw arrays?

In short, std::vector could avoid memory leak compare to raw array/pointer.

File: oap-ape/ape-native/src/reader.cc [hunk 1]
```diff
@@ -333,14 +338,10 @@ int Reader::readBatch(int32_t batchSize, int64_t* buffersPtr_, int64_t* nullsPtr
...
 
-  delete[] defLevel;
-  delete[] repLevel;
-  delete[] nullBitMap;
-
-  if (filterBufferCount > 0 || aggBufferCount > 0) {
-    delete[] buffersPtr;
-    delete[] nullsPtr;
-  }
+  // if (filterBufferCount > 0 || aggBufferCount > 0) {
```
Comment (winningsix) [inline oap-ape/ape-native/src/reader.cc]: Do we still need this?

**Review by winningsix — COMMENTED** [review]
One minor comment.

File: oap-ape/ape-native/src/reader.cc (hunk 1 above)
Comment (jikunshang) [reply oap-ape/ape-native/src/reader.cc]: removed.

**Pull request merged**